src/components/form-elements/TextInput.jsx

```diff
@@ -41,7 +41,8 @@ const TextInput = props => {
     'sg-input--invalid': valid === VALIDATION.INVALID,
     'sg-input--full-width': fullWidth,
     'sg-input--no-border': noBorder,
-    'sg-input--with-icon': withIcon
+    'sg-input--with-icon': withIcon,
+    'sg-search__input': type === 'search'
```

Comment:
Let's talk about it tomorrow. The issue is that, only in search, focus state of the input affects color of the icon. These elements are normally unrelated :/